src/Microsoft.Health.Dicom.SqlServer/Features/Partitioning/SqlPartitionStoreV6.cs

```diff
@@ -57,7 +57,7 @@ public override async Task<Partition> AddPartitionAsync(string partitionName, Ca
             }
             catch (SqlException ex)
             {
-                if (ex.Number == SqlErrorCodes.Conflict)
+                if (ex.Number == SqlErrorCodes.Conflict || ex.Number == SqlErrorCodes.UniqueKeyConstraintVoilation)
```

Comment:
If you are doing this we dont want the sql changes. We can make the sql changes in another PR

Reply:
Or if you would like it you can do it here and we dont have to check for unique constraint